test/integration/granting_permissions_test.rb

```diff
@@ -41,15 +41,20 @@ class GrantingPermissionsTest < ActionDispatch::IntegrationTest
     end
 
     should "not be able to grant permissions that are not grantable_from_ui" do
-      app = create(:application, name: "MyApp", with_non_delegatable_supported_permissions: %w[perm], with_non_delegatable_supported_permissions_not_grantable_from_ui: %w[user_update_permission])
+      app = create(
+        :application,
+        name: "MyApp",
+        with_non_delegatable_supported_permissions: %w[grantable_from_ui_perm],
+        with_non_delegatable_supported_permissions_not_grantable_from_ui: %w[not_grantable_from_ui_perm],
+      )
       @user.grant_application_signin_permission(app)
 
       visit edit_user_path(@user)
       click_link "Manage permissions"
       click_link "Update permissions for MyApp"
 
-      assert page.has_field?("perm")
-      assert page.has_no_field?("user_update_permission")
+      assert page.has_field?("grantable_from_ui_perm")
+      assert page.has_no_field?("not_grantable_from_ui_perm")
     end
   end
```

Comment:
The changes here and in the next test are purely for clarity. The third and fourth tests are the ones that are fixed. I'll update the commit message to reflect this